src/coreclr/pal/inc/unixasmmacrosamd64.inc

```diff
@@ -86,6 +86,12 @@ C_FUNC(\Name\()_End):
         .byte 0xE0
 .endm
 
+.macro TAILJMP_R10
+        .byte 0x49
+        .byte 0xFF
+        .byte 0xE2
+.endm
```

Comment:
This sequence has a superfluous rex prefix bit set to make sure the unwinder picks it up as an external tailcall:
https://github.com/dotnet/runtime/blob/21934674d16169dec4ed6b6d7a1092d12b7b354a/src/coreclr/vm/amd64/excepamd64.cpp#L368-L378

Reply:
This is Windows unwinder convention. I do not think it is applicable outside Windows.

Reply:
clang does not seem to generate a special jmp for tail calls on Unix:https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:36,endLineNumber:5,positionColumn:36,positionLineNumber:5,selectionStartColumn:36,selectionStartLineNumber:5,startColumn:36,startLineNumber:5),source:'extern+int+foo(int+x)%3B%0A%0Aextern+int+bar(int+x)%3B%0A%0Aint+foo(int+x)+%7B+return+bar(x%2B1)%3B+%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:48.26193742510047,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:2,lang:c%2B%2B,libs:!(),options:'-O3',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+clang+(trunk)+(Editor+%231)',t:'0')),header:(),k:51.73806257489954,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4

Reply:
I'll replace it with just `jmp r10`. I guess the `TAILJMP_RAX` uses in other non-Windows thunks can be cleaned up then.

Is there a difference between managed and native code here? The JIT is also generating these tailcalls with the extraneous rex prefixes. Can they be avoided outside Windows?

Reply:
Regular CoreCLR uses Windows unwinder outside Windows, so the JIT has to follow the Windows unwinder conventions there as well.

If we cared enough, we can skip the prefix for native AOT. We use native unwinder for native AOT so Windows unwinder conventions are not applicable outside Windows.